experimental/iterators/lib/Conversion/IteratorsToLLVM/IteratorsToLLVM.cpp

```diff
@@ -122,6 +148,66 @@ static Value getOrCreateGlobalString(OpBuilder &builder, Twine name,
                                globalPtr, ArrayRef<Value>({zero, zero}));
 }
 
+/// Lowers columnar_batch_from_memrefs to LLVM IR that extracts the bare
+/// pointers and the number of elements from the given memrefs.
+///
+/// Possible result:
+///
+/// %1 = builtin.unrealized_conversion_cast %0 :
+///        memref<3xi32> to !llvm.struct<(ptr<i32>, ptr<i32>, i64,
+///                                       array<1 x i64>, array<1 x i64>)>
+/// %2 = llvm.mlir.undef : !llvm.struct<(i64, ptr<i32>)>
+/// %3 = llvm.extractvalue %1[1] :
+///        !llvm.struct<(ptr<i32>, ptr<i32>, i64,
+///                      array<1 x i64>, array<1 x i64>)>
+/// %4 = llvm.extractvalue %1[3, 0] :
+///        !llvm.struct<(ptr<i32>, ptr<i32>, i64,
+///                      array<1 x i64>, array<1 x i64>)>
+/// %5 = llvm.insertvalue %3, %2[1 : index] : !llvm.struct<(i64, ptr<i32>)>
+/// %6 = llvm.insertvalue %4, %5[0 : index] : !llvm.struct<(i64, ptr<i32>)>
+struct ColumnarBatchFromMemrefsLowering
+    : public OpConversionPattern<ColumnarBatchFromMemrefsOp> {
+  ColumnarBatchFromMemrefsLowering(TypeConverter &typeConverter,
+                                   MLIRContext *context,
+                                   PatternBenefit benefit = 1)
+      : OpConversionPattern(typeConverter, context, benefit) {}
+
+  LogicalResult match(ColumnarBatchFromMemrefsOp op) const override {
+    return success();
+  }
+
+  void rewrite(ColumnarBatchFromMemrefsOp op, OpAdaptor adaptor,
```

Comment:
You want to check the memref is 1-D with a canonical layout; if not bail out.
Otherwise, surprises will occur if you ignore the non-0 offset or non-1 strides.

Reply:
ok I see it is in the verifier so please ignore then.